### PR TITLE
update LocaleApi and remove old metatdata access

### PR DESCRIPTION
**Pull request by craigh**
in preparation for migration away from ZLanguage and Locale.

| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | yes
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | #3365, #3328, #3180, https://github.com/zikula-modules/Multisites/issues/17
| License           | MIT
| Changelog updated | yes

## Description
update LocaleApi and remove old metatdata access in preparation for migration away from ZLanguage and ZLocale.

## Todos
- [x] Documentation
- [x] Changelog

### CODE REVIEW

**Comment by craigh** [comment]
wanted opinions on this approach before proceeding with further refactoring.

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Replacing ZLanguage
+===================
+
+In the examples below, the provided variable (e.g. $container, $kernel, etc) must be adjusted to match the environment
+from which it may be obtained. These variables are not defined explicitly in most cases.
+
+
+ZLanguage::getLanguageCode() becomes $container->getParameter('locale')
```
Comment (Guite) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: How about `$request->getLocale()`?

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+ZLanguage::getModuleDomain('AcmeFooModule') becomes $kernel->getModule('AcmeFooModule')->getTranslationDomain()
+
+ZLanguage::getDirection() becomes use `dir=auto` in the template instead
```
Comment (Guite) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: `{{ localeApi.language_direction }}` is deprecated then?

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+ZLanguage::getLanguageName($code) becomes \Intl::getLanguageBundle()->getLanguageName($locale)
+
+ZLanguage::getInstalledLanguages() becomes $localeApi->getSupportedLocales()
```
Comment (Guite) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: Great :smile:

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md (hunk 1 above)
Comment (Guite) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: Just saw `getLocale()` below.

What is the difference between `ZLanguage::getLanguageCode()` and `ZLanguage::getLocale()`?
What is the difference between `$container->getParameter('locale')` and `$request->getLocale()`?

The first is the default and the second one the current request's value only?

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+ZLanguage::getLocale() becomes $request->getLocale()
+
+ZLanguage::bind*Domain('AcmeFooModule') becomes no longer needed with symfony translator
+
+ZLanguage::getEncoding() becomes $kernel->getCharset()
+
+ZLanguage::isRequiredLangParam() is handled automatically by the router
+
+ZLanguage::countryMap() becomes \Intl::getRegionBundle()->getCountryNames()
```
Comment (Guite) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: There is also `\Intl::getRegionBundle()->getCountryName($countryCode)` available.

**Comment by Guite** [comment]
Please also add a pointer to http://symfony.com/doc/current/components/intl.html#accessing-icu-data to the docs.

**Comment by Guite** [comment]
I like the direction where this is heading to.

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md (hunk 1 above)
Comment (craigh) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: I suppose so. not sure.

File: src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md (hunk 2 above)
Comment (craigh) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: not deprecated - removed. broken. see also #3365

Comment (craigh) [inline src/docs/Core-2.0/TranslationAndLanguage/MigrationFromZLanguage.md]: I suppose we could provide some BC

**Comment by craigh** [comment]
@Guite did you also notice how I changed the scan for locales? It is not using the `locale.ini` file any longer, but the provided translations. ([option 2](https://github.com/zikula/core/issues/3328#issuecomment-270009584) from #3328)

**Comment by Guite** [comment]
This is fine I think.

**Pull request merged**